src/transforms/route.rs

```diff
@@ -38,15 +39,17 @@ impl SyncTransform for Route {
         event: Event,
         output: &mut vector_core::transform::TransformOutputsBuf,
     ) {
+        let mut passed: usize = 0;
         for (output_name, condition) in &self.conditions {
             if condition.check(&event) {
                 output.push_named(output_name, event.clone());
             } else {
-                emit!(&RouteEventDiscarded {
```

Comment:
Should we document (in the upgrade guide perhaps) that this metric is disappearing for `route`?